lib/vector-core/src/byte_size_of.rs

```diff
@@ -68,3 +68,26 @@ where
         self.as_ref().map_or(0, |x| x.allocated_bytes())
     }
 }
+
+macro_rules! num {
+    ($t:ty) => {
+        impl ByteSizeOf for $t {
+            fn allocated_bytes(&self) -> usize {
+                0
+            }
+        }
+    };
+}
```

Comment:
Why are we counting primitive numeric types as 0 bytes?

Reply:
Quirk of how `ByteSizeOf` works. They have 0 _allocated_ bytes, which ends up being added to their `mem::size_of` size.

Reply:
Ohhhhh, right.  👍🏻